src/core/utils/useNotificationMessage.ts

```diff
@@ -0,0 +1,32 @@
+import { useState } from "react";
+
+type UseNotificationOptionsType = {
+  timeout?: number;
+  scrollToTop?: boolean;
+};
+type UseNotificationReturnType = [string | null, (text: string) => void];
+
+export const useNotificationMessage = ({
+  timeout,
+  scrollToTop
+}: UseNotificationOptionsType): UseNotificationReturnType => {
+  const [notificationMessage, setNotificationMessage] = useState<string | null>(
+    null
+  );
+
+  const handleNotificationMessage = (text: string) => {
+    setNotificationMessage(text);
+    if (scrollToTop) {
+      window.scrollTo(0, 0);
+    }
+    if (timeout) {
+      setTimeout(() => {
+        setNotificationMessage(null);
+      }, timeout);
+    }
+  };
+
+  return [notificationMessage, handleNotificationMessage];
```

Comment:
I think you should return the PromoBar instead of the notificationMessage. Then we make the whole thing more declarative and nizzle.